Lib/test/test_ctypes/test_arrays.py

```diff
@@ -23,6 +27,25 @@ def ARRAY(*args):
 
 
 class ArrayTestCase(unittest.TestCase):
+    def test_inheritance_hierarchy(self):
+        self.assertEqual(_CData.__name__, "_CData")
+        self.assertEqual(Array.mro(), [Array, _CData, object])
+
+        self.assertEqual(ArrayType.__name__, "PyCArrayType")
+        self.assertEqual(type(ArrayType), type)
+
+    def test_instantiation(self):
+        with self.assertRaisesRegex(TypeError, "abstract class"):
+            Array()
+
+        ArrayType("Foo", (Array,), {"_length_": 1, "_type_": c_int})
```

Comment:
IIRC, we have a test helper in `test.support` for this. `git grep disallow_instantiation` (from the top of my head).

Reply:
I had taken a look at the `disallow_instantion`, but found that it assumes the error message must have the exact format https://github.com/python/cpython/blob/f19b93fce04fb0bc9b59071915a6aa6b01860d8a/Lib/test/support/__init__.py#L2083.

I think the error is raised by Python runtime, if the `Py_TPFLAGS_DISALLOW_INSTANTIATION` flag is set, so the error message has a fixed format.

In Our situation, the error message is "abstract class", I found it's raised in the type's `__new__` method, so whether we set the `Py_TPFLAGS_DISALLOW_INSTANTIATION` flag or not, the error will be raised anyway.

So can we use another method to check we kept the same behavior after the change, like check the type's `__flags__`
 field, which is just the `tp_flags` slot in C? The mutability check can be done this way too.

Reply:
Using `__flags__` is nice; I wonder why we didn't do that in the first place 😆 Perhaps we should improve `test.support.disallow_instantiation` to check `__flags__` as well!

Anyway, there is precedent for your suggestion in `Lib/test/test_call.py`. Go ahead!